src/ssd/ssd_titlebar.c

```diff
@@ -137,11 +147,16 @@ set_squared_corners(struct ssd *ssd, bool enable)
 			struct ssd_button *button = node_ssd_button_from_node(part->node);
 
 			/* Toggle background between invisible and titlebar background color */
-			struct wlr_scene_rect *rect = wlr_scene_rect_from_node(button->background);
-			wlr_scene_rect_set_color(rect, !enable ? (float[4]) {0, 0, 0, 0} : (
-				subtree == &ssd->titlebar.active
-					? rc.theme->window_active_title_bg_color
-					: rc.theme->window_inactive_title_bg_color));
+			struct wlr_scene_rect *rect = lab_wlr_scene_get_rect(button->background);
+			/*Check for custom color as well*/
+			float custom_color[4];
+			bool is_custom_color_available =
+				window_rules_get_custom_border_color(ssd->view, custom_color);
+			wlr_scene_rect_set_color(rect, is_custom_color_available ? custom_color
+				: (!enable ? (float[4]) {0, 0, 0, 0}
+					: (subtree == &ssd->titlebar.active
+						? rc.theme->window_active_title_bg_color
+						: rc.theme->window_inactive_title_bg_color)));
```

Comment:
This is getting really hard to parse; a separate function would probably be a better option here.

Reply:
if statement added, the rest of code turned back.